src/engine/enginebuffer.cpp

```diff
@@ -430,6 +430,11 @@ void EngineBuffer::requestSyncMode(SyncMode mode) {
     }
 }
 
+void EngineBuffer::requestSyncPosition() {
+    m_iEnableSyncQueued = SYNC_REQUEST_POSITION;
+    m_iSyncModeQueued = SYNC_INVALID;
```

Comment:
using two atomic variables in `EngineBuffer` seems like a race condition. This should be replaced by two vars in a struct with an atomic guard. Thoughts?